mesonbuild/linkers/detect.py

```diff
@@ -72,13 +72,14 @@ def guess_win_linker(env: 'Environment', compiler: T.List[str], comp_class: T.Ty
     override = []  # type: T.List[str]
     value = env.lookup_binary_entry(for_machine, comp_class.language + '_ld')
     if value is not None:
+        compiler = value # If meson is explicitly told a value for the linker so use it!
```

Comment:
If meson is explicitly told a value for the linker, use it!

Reply:
This change makes me nervous, this should only get hit in the case of gcc-esque clang, which expects to get the linker value as `-fuse-ld=...`, so I think this is going to break things.